Tools/CMake/AMReX_Options.cmake

```
@@ -90,15 +90,33 @@ endif ()
 print_option( DIM )
 
 #
-# Programming models  ========================================================
+# Parallel backends    ========================================================
 #
-option( ENABLE_MPI  "Enable MPI"  ON)
+
+# For the time being ENABLE_DPCPP is defined before project() is called
+# Check whether the C++ compiler is dpcpp
+print_option(ENABLE_DPCPP)
+if (ENABLE_DPCPP AND (NOT (CMAKE_CXX_COMPILER MATCHES "dpcpp") ) )
+   message(FATAL_ERROR "\nENABLE_DPCPP=${ENABLE_DPCPP} but CXX compiler is not dpcpp\n")
+endif ()
+
+cmake_dependent_option( ENABLE_DPCPP_AOT  "Enable DPCPP AOT"  OFF
```

I think this does not really work with the compiler yet:
```suggestion
cmake_dependent_option( ENABLE_DPCPP_AOT  "Enable DPCPP ahead-of-time compilation (WIP)"  OFF
```

---

Should we issue a warning in case a user sets it? This variable is present in the GNUMake so I would like to keep it in the CMake too even if not yet usable.

---

Good idea, let's issue a warning for now.